### PR TITLE
broker: drop -k-ary option, rename tbon.arity attribute to tbon.fanout

### PR DESCRIPTION
**Pull request by garlick**
As noted in #3795, it is confusing that the `tbon.arity` attribute can only be changed with the broker `--k-ary` option.  Also we seem to be using "fanout", "arity", and "k" for the same thing.

Drop the option, make the attribute settable, and rename the attribute to `tbon.fanout` (which seems to me the most clear of the three names).

This will require a fluxion update.

### CODE REVIEW

File: etc/flux.service.in [hunk 1]
```diff
@@ -7,7 +7,7 @@ TimeoutStopSec=90
 KillMode=mixed
 ExecStart=@X_BINDIR@/flux broker \
   --config-path=@X_SYSCONFDIR@/flux/system/conf.d \
-  --k-ary=256 \
+  -Stbon.fanout=256 \
```
Comment (chu11) [inline etc/flux.service.in:10]: should the commit message header be "rename tbon.arity"?   Preferable is "rename tbon.arity to tbon.fanout", but I assume that would make the line too long.

**Review by chu11 — APPROVED** [review]
LGTM, just a commit message comment.  I also noticed "arity" is in the doc spelling file, dunno if we want to remove it there just in case we ever try to use it again in a doc.

File: etc/flux.service.in (hunk 1 above)
Comment (garlick) [reply etc/flux.service.in:10]: Heh, in the past I've felt the new name would have more significance than the old one in the history.  In this case, I think maybe a generic message that doesn't mention names is probably better.  Will psuh an update.

**Comment by garlick** [comment]
Force pushed a change that drops "arity" from the dictonary and tweaks the commit messages slightly per @chu11.

**Comment by chu11** [comment]
LGTM

**Comment by garlick** [comment]
> This will require a fluxion update.

Huh, I may have been confused.  Fluxion is working OK with this change.

**Comment by garlick** [comment]
Coverage test got canceled after 30m.  Restarting.

**Comment by grondo** [comment]
> Coverage test got canceled after 30m. Restarting.

I have noticed that sometimes 30m is not quite long enough for coverage tests. We may have to bump this time limit up soon.

**Review by grondo — APPROVED** [review]
LGTM!

**Pull request merged**